src/paperbot/infrastructure/swarm/agents/knowledge_manager.py

```
@@ -0,0 +1,147 @@
+            lines.append("- (no conventions extracted yet)")
+        return "\n".join(lines)
+
+    def _build_learnings(self, tasks: List["AgentTask"]) -> str:
+        lines = [
+            "# Learnings",
+            "",
+            "Insights from task execution:",
+            "",
+        ]
+        if self.commander.wisdom.learnings:
+            for l in self.commander.wisdom.learnings:
```

Ruff (0.15.5)

[error] Ambiguous variable name: `l`

(E741)